### PR TITLE
Np 38271 aggregation for ticket status

### PR DESCRIPTION
**Pull request by EyobSikt**
- Added status to the aggregations. 
- Removed restriction from status. 
- All status shows.

### CODE REVIEW

File: search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java [hunk 1]
```diff
@@ -466,6 +465,114 @@ void searchTicketsResponseShouldFormatAggregationsCorrectly() throws IOException
         assertThat(aggregations, is(Matchers.equalTo(expected)));
     }
 
+    @Test
+    void shouldSendQueryWithAllNeededClauseForPublishingRequestTypeWhenSearchingForTickets()
```
Comment (brinxmat) [inline search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java:469]: Clauses

File: search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java [hunk 2]
```diff
@@ -466,6 +465,114 @@ void searchTicketsResponseShouldFormatAggregationsCorrectly() throws IOException
...
+                .anyMatch(condition -> condition.fieldName().equals("status"));
+        var mustExistsOrganizationIdsInDoiRequests =
+            rulesForIncludingExistingPublicationConversation.stream()
+                .anyMatch(condition -> condition.fieldName().equals(
+                    "organizationIds"));
+        assertTrue(mustExistsTicketStatusInDoiRequests, "Could not find rule for including ticket status for "
+                                                        + "GeneralSupportCase");
+        assertTrue(mustExistsOrganizationIdsInDoiRequests,
+                   "Could not find rule for including OrganizationIds for GeneralSupportCase");
+    }
```
Comment (brinxmat) [inline search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java:574]:
```
        var rulesForIncludingPublicationConversation = extractQueryBuilderValues(sentRequest);

        assertThat(rulesForIncludingPublicationConversation, hasItem(GENERAL_SUPPORT_CASE));
    }

    private List<Object> extractQueryBuilderValues(SearchRequest sentRequest) {
        return listAllInclusionAndExclusionRulesForPublicationConversation(sentRequest)
                .stream().map(MatchQueryBuilder::value)
                .collect(Collectors.toList());
    }
```

File: search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java [hunk 3]
```diff
@@ -466,6 +465,114 @@ void searchTicketsResponseShouldFormatAggregationsCorrectly() throws IOException
...
+                .anyMatch(rule -> rule.value().equals(PUBLISHING_REQUEST));
+        var mustIncludePublishingRequestForDraftPublications =
+            rulesForIncludingPublishingRequest.stream().anyMatch(rule -> rule.value().equals(
+                DRAFT_PUBLICATION_STATUS));
+
+        assertTrue(mustIncludePublishingRequestType,
+                   "Could not find rule for including PublishingRequest");
+        assertTrue(mustIncludePublishingRequestForDraftPublications,
+                   "Could not find rule for including for Draft Publications");
+    }
```
Comment (brinxmat) [inline search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java:502]: Use Hamcrest, see below

File: search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java (hunk 1 above)
Comment (brinxmat) [inline search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java:469]: I'm unsure about the naming…

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Review by ketilaa — CHANGES_REQUESTED** [review]
Have a look at comments from @brinxmat

File: search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java (hunk 1 above)
Comment (EyobSikt) [reply search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java:407]: changed to Rules.

File: search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java [hunk 4]
```diff
@@ -466,6 +465,114 @@ void searchTicketsResponseShouldFormatAggregationsCorrectly() throws IOException
...
+        AtomicReference<SearchRequest> sentRequestBuffer = new AtomicReference<>();
+        var restClientWrapper = new RestHighLevelClientWrapper((RestHighLevelClient) null) {
+            @Override
+            public SearchResponse search(SearchRequest searchRequest, RequestOptions requestOptions) {
+                sentRequestBuffer.set(searchRequest);
+                return mockSearchResponse;
+            }
+        };
+
+        var searchClient = new SearchClient(restClientWrapper, cachedJwtProvider);
```
[the remark this answers is not in the document]
Comment (EyobSikt) [reply search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java:420]: Done. This removes also duplicate code. (A very good suggestion.)

File: search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java (hunk 1 above)
Comment (EyobSikt) [reply search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java:407]: Done. Hope it gives now meaning.

File: search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java (hunk 2 above)
Comment (EyobSikt) [reply search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java:512]: > Have a look at comments from @brinxmat

@ketilaa fixed the comments from @brinxmat. Waiting for RE-request revivew.

**Review by ketilaa — DISMISSED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**